### PR TITLE
Allow null value for CheckboxListTile

### PR DESCRIPTION
**Pull request by woprandi**
## Description

Allow null value through tristate property for CheckboxListTile (exactly like Checkbox).
The docstring has been copied from the `Checkbox` class

## Related Issues

https://github.com/flutter/flutter/issues/58108

## Tests

I added the following tests:
* Setting new `tristate` property to true and tap few times to change value. Exactly like` Checkbox tristate test 
`
## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/checkbox_list_tile.dart [hunk 1]
```diff
@@ -268,7 +268,9 @@ class CheckboxListTile extends StatelessWidget {
     this.controlAffinity = ListTileControlAffinity.platform,
     this.autofocus = false,
     this.contentPadding,
-  }) : assert(value != null),
+    this.tristate = false,
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/checkbox_list_tile.dart:271]: Can you please update the constructor documentation to describe that `tristate` can not be null, and that `value` can be null if `tristate` is true?

**Review by darrenaustin — CHANGES_REQUESTED** [review]
Other than the one doc request, this looks great.

File: packages/flutter/lib/src/material/checkbox_list_tile.dart (hunk 1 above)
Comment (woprandi) [reply packages/flutter/lib/src/material/checkbox_list_tile.dart:272]: Done. BTW the prohibition of null value for `isTreeLine`, `selected`, `controlAffinity` and `autofocus` parameters is also missing. But I think this PR is not the right place to fix that.

**Review by darrenaustin — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**